tests/net/traffic_class/src/main.c

```diff
@@ -968,46 +969,42 @@ static void test_traffic_class_recv_data_mix_all_2(void)
-			 ztest_unit_test(test_traffic_class_send_data_prio_ee),
-			 ztest_unit_test(test_traffic_class_send_data_prio_ca),
-			 ztest_unit_test(test_traffic_class_send_data_prio_vi),
-			 ztest_unit_test(test_traffic_class_send_data_prio_vo),
-			 ztest_unit_test(test_traffic_class_send_data_prio_ic),
-			 ztest_unit_test(test_traffic_class_send_data_prio_nc),
-			 /* Then mix traffic classes and verify that higher
-			  * class packets are sent first.
-			  */
-			 ztest_unit_test(test_traffic_class_send_data_mix),
-			 ztest_unit_test(test_traffic_class_send_data_mix_all_1),
-			 ztest_unit_test(test_traffic_class_send_data_mix_all_2),
-			 ztest_unit_test(test_traffic_class_cleanup_tx),
-
-			 /* Same tests for received packets */
-			 ztest_unit_test(test_traffic_class_setup_rx),
-			 ztest_unit_test(test_traffic_class_setup_recv),
-			 ztest_unit_test(test_traffic_class_recv_data_prio_bk),
-			 ztest_unit_test(test_traffic_class_recv_data_prio_be),
-			 ztest_unit_test(test_traffic_class_recv_data_prio_ee),
-			 ztest_unit_test(test_traffic_class_recv_data_prio_ca),
-			 ztest_unit_test(test_traffic_class_recv_data_prio_vi),
-			 ztest_unit_test(test_traffic_class_recv_data_prio_vo),
-			 ztest_unit_test(test_traffic_class_recv_data_prio_ic),
-			 ztest_unit_test(test_traffic_class_recv_data_prio_nc),
-			 ztest_unit_test(test_traffic_class_recv_data_mix),
-			 ztest_unit_test(test_traffic_class_recv_data_mix_all_1),
-			 ztest_unit_test(test_traffic_class_recv_data_mix_all_2),
-			 ztest_unit_test(test_traffic_class_cleanup_rx)
-			 );
-
-	ztest_run_test_suite(net_traffic_class_test);
-}
+ZTEST(traffic_class_test_suite, test_traffic_class_tx_rx)
+{
+	test_traffic_class_setup_tx();
+	/* Send only same priority packets and verify that
+	 * all are sent with proper traffic class.
+	 */
+	test_traffic_class_send_data_prio_bk();
+	test_traffic_class_send_data_prio_be();
+	test_traffic_class_send_data_prio_ee();
+	test_traffic_class_send_data_prio_ca();
+	test_traffic_class_send_data_prio_vi();
+	test_traffic_class_send_data_prio_vo();
+	test_traffic_class_send_data_prio_ic();
+	test_traffic_class_send_data_prio_nc();
+	/* Then mix traffic classes and verify that higher
+	 * class packets are sent first.
+	 */
+	test_traffic_class_send_data_mix();
+	test_traffic_class_send_data_mix_all_1();
+	test_traffic_class_send_data_mix_all_2();
+	test_traffic_class_cleanup_tx();
+
+	test_traffic_class_setup_rx();
+	test_traffic_class_setup_recv();
+	test_traffic_class_recv_data_prio_bk();
+	test_traffic_class_recv_data_prio_be();
+	test_traffic_class_recv_data_prio_ee();
+	test_traffic_class_recv_data_prio_ca();
+	test_traffic_class_recv_data_prio_vi();
+	test_traffic_class_recv_data_prio_vo();
+	test_traffic_class_recv_data_prio_ic();
+	test_traffic_class_recv_data_prio_nc();
+	test_traffic_class_recv_data_mix();
+	test_traffic_class_recv_data_mix_all_1();
+	test_traffic_class_recv_data_mix_all_2();
+	test_traffic_class_cleanup_rx();
```

Comment:
Are these all inter-dependent? Each test should be hermetic and run individually in any order.

Reply:
These supposed to be independent, but actually not.
I have modified some code and made them independent. Thanks to review.